backend/src/procedure/js_scripts.rs

```diff
@@ -605,8 +615,49 @@ mod js_runtime {
             let ctx = state.borrow::<JsContext>();
             Ok(ctx.sandboxed)
         }
+
+        /// We need to make sure that during the file accessing, we don't reach beyond our scope of control
+        fn is_subset(parent: impl AsRef<Path>, child: impl AsRef<Path>) -> bool {
```

Comment:
If you have a better solution, that would be lovely. It is a marked improvement from the previous.

Reply:
But will bring back the canoncialize since there is a possibility that one could create a link, and following that link would allow modification of another volume, maybe. But I don't want to take that chance so using the canonicalize now

Reply:
This looks *more* permissive than before. You're checking prefixes of path segments, doesn't that mean /foo/baz would match against /foobar/baz? That doesn't make sense to me.